include/nbl/asset/utils/CSPIRVIntrospector.h

```diff
@@ -526,6 +527,30 @@ class NBL_API2 CSPIRVIntrospector : public core::Uncopyable
 
 					return range;
 				}
+				inline SDescriptorArrayInfo addDescriptorCount(const uint32_t size, const bool sizeIsLiteral)
+				{
+					const auto descCount = alloc<SDescriptorArrayInfo>(1u);
+					auto descriptorArraySize = descCount(m_memPool.data());
+
+					// the API for this spec constant checking is truly messed up
+					if (size == 0u)
+					{
+						descriptorArraySize[0].count = size;
+						descriptorArraySize[0].countMode = SDescriptorArrayInfo::DESCRIPTOR_COUNT::RUNTIME;
+					}
+					else if (sizeIsLiteral)
```

Comment:
`sizeIsLiteral` is the first thing you should check, because right not spec constant ID of 0 would falsely register as a runtime array

```cpp
if (sizeIsLiteral)
{
    descriptorArraySize.count = size;
    descriptorArraySize.countMode = size!=0 ? STATIC:RUNTIME;
}
else
{
    descriptorArraySize.count = size;
    descriptorArraySize.countMode = SPEC_CONSTANT;
}
```